### PR TITLE
Use docker only in tests

### PR DESCRIPTION
**Pull request by jajik**

### CODE REVIEW

**Review by jfclere — CHANGES_REQUESTED** [review]

File: test/native/tests.sh [hunk 1]
```diff
@@ -146,27 +154,19 @@ starttomcat() {
...
-	    echo "docker run failed"
+	    echo "podman run failed"
 	    exit 1
     fi
 }
 
 # Start the webapp on the given tomcat
 # wait for the tomcat to start.
 startwebapptomcat() {
-    while true
```
[the remark this answers is not in the document]
Comment (jajik) [reply test/native/tests.sh:161]: Oh, now I get the intention. But then there should not be the `-a` argument for `podman ps`, I think...

**Review by rhusar — CHANGES_REQUESTED** [review]
Please let's NOT do this - make the container binary switchable / configurable / autodetected.

Instead of this you could just update the readme to to install podman-docker package or hell, run 'alias docker=podman'

**Comment by jajik** [comment]
> Please let's NOT do this - make the container binary switchable / configurable / autodetected.
> 
> Instead of this you could just update the readme to to install podman-docker package or hell, run 'alias docker=podman'

Personally, I don't think the change is necessary – previously, the podman was used for the httpd's image, docker for tomcat. However, with the latest changes (9096b14f1c94be6f08aebe4e642a7e267ade67f6, e4422b6dd9c6d7255cf2d6f0ce422b127c7c3709) the tests are broken and we should address it. (The second commit suggest that we want to move to podman, hence this PR.)

**Comment by rhusar** [comment]
> > Please let's NOT do this - make the container binary switchable / configurable / autodetected.
> > Instead of this you could just update the readme to to install podman-docker package or hell, run 'alias docker=podman'
> 
> Personally, I don't think the change is necessary – previously, the podman was used for the httpd's image, docker for tomcat. However, with the latest changes ([9096b14](https://github.com/modcluster/mod_proxy_cluster/commit/9096b14f1c94be6f08aebe4e642a7e267ade67f6), [e4422b6](https://github.com/modcluster/mod_proxy_cluster/commit/e4422b6dd9c6d7255cf2d6f0ce422b127c7c3709)) the tests are broken and we should address it. (The second commit suggest that we want to move to podman, hence this PR.)

Yeah, these went in without a pull request - it looks like @jfclere accidentally pushed directly to the branch so this was missed in review.

**Comment by jajik** [comment]
I changed the podman to docker as agreed. Also, I made a few changes to tests. It seems that the previously introduced changes are causing issues when tests are re-run; one has to be sure that everything from the previous run is cleaned up (tests are now faster so it seems like a reasonable trade off until we rework the tests in the future).

**Review by rhusar — APPROVED** [review]
LGTM now.

**Pull request merged**